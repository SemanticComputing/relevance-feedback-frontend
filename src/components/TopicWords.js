import React from 'react';
import { array } from 'prop-types';
import { map, min } from 'lodash';

const TopicWords = ({ topic }) => {

  const getSize = (word) => min([8 + word[1] * 1000, 60]);

  const words = map(topic, (word) => (
    <span key={word[0]} style={{ fontSize: getSize(word) }}>
      {word[0]}{' '}
    </span>
  ));

  return (
    <div>
      {words}
    </div>
  );
};

TopicWords.propTypes = {
  topic: array
};

export default TopicWords;
