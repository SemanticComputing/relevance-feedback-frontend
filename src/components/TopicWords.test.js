import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TopicWords from './TopicWords';

const topic = [
  ['word', 0.2],
  ['something', 0.1489],
  ['innovation', 0.1],
  ['trash', 0.01],
];

describe('TopicWords', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TopicWords topic={topic} />, div);
  });

  it('displays topic words', () => {
    const tree = renderer
      .create(<TopicWords topic={topic} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
