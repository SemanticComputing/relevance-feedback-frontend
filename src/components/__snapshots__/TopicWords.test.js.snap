// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicWords displays topic words 1`] = `
<div>
  <span
    style={
      Object {
        "fontSize": 260,
      }
    }
  >
    word
     
  </span>
  <span
    style={
      Object {
        "fontSize": 193.57,
      }
    }
  >
    something
     
  </span>
  <span
    style={
      Object {
        "fontSize": 130,
      }
    }
  >
    innovation
     
  </span>
  <span
    style={
      Object {
        "fontSize": 13,
      }
    }
  >
    trash
     
  </span>
</div>
`;
